services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/nodeports.py

```diff
@@ -243,14 +248,32 @@ def _is_zip_file(file_path: Path) -> bool:
 _shutil_move = aiofiles.os.wrap(shutil.move)
 
 
+async def _move_file_to_input_port(
+    final_path: Path, downloaded_file: Path, dest_folder: PrunableFolder
+) -> None:
+    _logger.debug("moving %s", downloaded_file)
+    final_path = final_path / downloaded_file.name
+
+    # ensure parent exists
+    final_path.parent.mkdir(exist_ok=True, parents=True)
+
+    await _shutil_move(downloaded_file, final_path)
+
+    # NOTE: after the download the current value of the port
+    # makes sure previously downloaded files are removed
+    dest_folder.prune(exclude={final_path})
```

Comment:
this NOTE does not mean something I can understand...
why not using a context manager if you need to clean stuff?